Applications/SlicerApp/qSlicerAppMainWindow.cxx

```diff
@@ -154,39 +156,80 @@ void qSlicerAppMainWindowPrivate::setupUi(QMainWindow * mainWindow)
-  this->HelpMenu->addAction(helpAboutSlicerAppAction);
+  this->HelpMenu->addAction(this->HelpAboutSlicerAppAction);
 
   //----------------------------------------------------------------------------
   // Icons in the menu
   //----------------------------------------------------------------------------
   // Customize QAction icons with standard pixmaps
-  QIcon networkIcon = mainWindow->style()->standardIcon(QStyle::SP_DriveNetIcon);
-  QIcon informationIcon = mainWindow->style()->standardIcon(QStyle::SP_MessageBoxInformation);
-  QIcon questionIcon = mainWindow->style()->standardIcon(QStyle::SP_MessageBoxQuestion);
+  this->updateIconPalette();
+  setThemeIcon(this->HelpAboutSlicerAppAction, "help-about");
+  setThemeIcon(this->HelpReportBugOrFeatureRequestAction, "tools-report-bug");
+}
 
-  helpAboutSlicerAppAction->setIcon(informationIcon);
-  helpReportBugOrFeatureRequestAction->setIcon(questionIcon);
+//-----------------------------------------------------------------------------
+void qSlicerAppMainWindowPrivate::updateIconPalette()
+{
+  QPalette palette = qSlicerApplication::application()->palette();
+  QStyle* lightStyle = QStyleFactory::create("Light Slicer");
+  QString hexColor;
+  if (palette == lightStyle->standardPalette())
+  {
+    hexColor = lightStyle->standardPalette().color(QPalette::Text).name();
+  }
+  else
+  {
+    QStyle* darkStyle = QStyleFactory::create("Dark Slicer");
+    hexColor = darkStyle->standardPalette().color(QPalette::Text).name();
+  }
+  this->HelpDocumentationAction->setIcon(this->getColorizedIcon(hexColor, ":/Icons/Scalable/OpenExternal.svg"));
+  this->HelpKeyboardShortcutsAction->setIcon(this->getColorizedIcon(hexColor, ":/Icons/Scalable/Keyboard.svg"));
+  this->HelpVisitSlicerForumAction->setIcon(this->getColorizedIcon(hexColor, ":/Icons/Scalable/Forum.svg"));
+  this->HelpReportBugOrFeatureRequestAction->setIcon(this->getColorizedIcon(hexColor, ":/Icons/Scalable/Warning.svg"));
+  this->HelpAboutSlicerAppAction->setIcon(this->getColorizedIcon(hexColor, ":/Icons/Scalable/Info.svg"));
+}
 
-  setThemeIcon(helpAboutSlicerAppAction, "help-about");
-  setThemeIcon(helpReportBugOrFeatureRequestAction, "tools-report-bug");
+//-----------------------------------------------------------------------------
+QIcon qSlicerAppMainWindowPrivate::getColorizedIcon(const QString& hexColor, const QString& resourcePath)
+{
+  QIcon icon;
+  if (hexColor == "#000000")
+  {
+    // resource icons are already colored black by default
+    icon = QIcon(resourcePath);
+  }
+  else
+  {
+    QFile file(resourcePath);
+    file.open(QIODevice::ReadOnly);
+    QByteArray data = file.readAll();
+    QString newHexString = "fill=\"" + hexColor;
+    QByteArray newByteArray = newHexString.toLocal8Bit();
+    const char *newHexChar = newByteArray.data();
+    data.replace("fill=\"#000000", newHexChar);
+    QPixmap pixmap;
+    pixmap.loadFromData(data);
+    icon = QIcon(pixmap);
+  }
+  return icon;
 }
```

Comment:
It seems like this basic code structure gets repeated a lot - is there a way to factor it out?

Reply:
In this PR (or a follow-up one), adding a utility function called `getColorizedIcon(const QString& hexColor, const QString& fileName)` to https://github.com/commontk/CTK/blob/master/Libs/Widgets/ctkWidgetsUtils.h may be sensible.

Reply:
Corresponding PR https://github.com/commontk/CTK/pull/1040